Yank/experiment.py

```diff
@@ -700,16 +705,24 @@ def run_experiments(self):
 
             # Cycle between experiments every switch_experiment_interval iterations
             # until all of them are done.
-            completed = [False for _ in range(len(all_experiments))]
-            while not all(completed):
+            nand = [False for _ in range(len(all_experiments))]
+            while len(all_experiments) > 0:
                 # Distribute experiments across MPI communicators if requested
+                completed = [False] * len(all_experiments)
                 if processes_per_experiment is None:
                     for exp_index, exp in enumerate(all_experiments):
                         completed[exp_index] = self._run_experiment(exp)
                 else:
-                    completed, experiment_indices = mpi.distribute(self._run_experiment,
-                                                                   distributed_args=all_experiments,
-                                                                   group_nodes=processes_per_experiment)
+                    completed = mpi.distribute(self._run_experiment,
+                                               distributed_args=all_experiments,
+                                               group_nodes=processes_per_experiment,
+                                               send_results_to='all')
+                # Remove any completed experiments, releasing possible parallel resources to be reused
+                for exp_index in range(len(all_experiments)):
```

Comment:
You may have to perform the loop in reverse order or you'll delete the wrong experiments when multiple are completed.